packages/js-moi-providers/src.ts/base-provider.ts

```diff
@@ -1277,19 +1272,14 @@ export class BaseProvider extends AbstractProvider {
             ErrorUtils.throwError("Invalid response received", ErrorCode.SERVER_ERROR);
         }
 
-        if (typeof event === "string" && ["newTesseracts"].includes(event)) {
+        if (typeof event === "string" && event === "newTesseracts") {
             return result;
         }
 
-        if (typeof event === "object" && event.event === "newTesseractsByAccount") {
+        if (typeof event === "object" && ["newTesseractsByAccount", "newLogs"].includes(event.event)) {
             return result;
         }
 
-        if (typeof event === "object" && event.event === "newLogs") {
-            const log = result as Log;
-            return { ...log, data: encodeToString(decodeBase64(log.data)) };
-        }
-
         ErrorUtils.throwArgumentError("Invalid event type", "event", event);
     }
 
```

Comment:
can we improve it if possible

Reply:
https://github.com/sarvalabs/js-moi-sdk/pull/94/commits/6459ddc5528fba8d81e6bac2055ee211f79121a0